jwplatform/api.js

```diff
@@ -36,8 +36,8 @@ class JWPlatformAPI {
             upload_method: 'single',
         });
         return this.videos.create(videoData).then(response => {
-            const { path, protocol, address } = response.link;
-            const uploadUrl = `${protocol}://${address}${path}`;
+            const { path, address } = response.link;
+            const uploadUrl = `https://${address}${path}`;
```

Comment:
This prevents internal use of the client, and ideally, the API itself should never be returning `http` as the protocol.